### PR TITLE
Support deployment in add-to-mesh and remove-from-mesh

### PR DESCRIPTION
**Pull request by irisdingbj**
resolves https://github.com/istio/istio/issues/16997

### CODE REVIEW

File: istioctl/cmd/add-to-mesh_test.go [hunk 1]
```diff
@@ -126,13 +126,19 @@ var (
 func TestAddToMesh(t *testing.T) {
 	cases := []testcase{
 		{
-			description:       "Invalid command args",
+			description:       "Invalid command args -missing service name",
```
Comment (esnible) [inline istioctl/cmd/add-to-mesh_test.go]: Add space between `-` and `missing` similar to line 135.

File: istioctl/cmd/add-to-mesh_test.go [hunk 2]
```diff
@@ -143,6 +149,18 @@ func TestAddToMesh(t *testing.T) {
...
 		},
+		{
+			description: "valid case - add deployment into mesh",
+			args: strings.Split("experimental add-to-mesh deployment details-v1 --meshConfigFile testdata/mesh-config.yaml"+
+				" --injectConfigFile testdata/inject-config.yaml"+
+				" --valuesFile testdata/inject-values.yaml", " "),
+			expectedException: false,
+			k8sConfigs:        cannedK8sConfigs,
+			expectedOutput: "deployment details-v1.default updated successfully with Istio sidecar injected.\n" +
+				"Next Step: Add related labels to the deployment to align with Istio's requirement: " +
```
Comment (esnible) [inline istioctl/cmd/add-to-mesh_test.go]: `Add 'app' and 'version' labels for Istio telemetry:`

File: istioctl/cmd/add-to-mesh_test.go [hunk 3]
```diff
@@ -152,6 +170,15 @@ func TestAddToMesh(t *testing.T) {
 			k8sConfigs:        cannedK8sConfigs,
 			expectedOutput:    "Error: services \"test\" not found\n",
 		},
+		{
+			description: "deployment not exists",
```
Comment (esnible) [inline istioctl/cmd/add-to-mesh_test.go]: `"deployment does not exist"`

File: istioctl/cmd/add-to-mesh_test.go (hunk 1 above)
Comment (irisdingbj) [reply istioctl/cmd/add-to-mesh_test.go]: fixed

File: istioctl/cmd/add-to-mesh_test.go (hunk 2 above)
Comment (irisdingbj) [reply istioctl/cmd/add-to-mesh_test.go]: This is actually coming from original add-to-mesh command.

File: istioctl/cmd/add-to-mesh_test.go (hunk 3 above)
Comment (irisdingbj) [reply istioctl/cmd/add-to-mesh_test.go]: fixed

**Comment by irisdingbj** [comment]
@esnible PTAL

**Pull request merged**